src/api/Bazaar.ts

```diff
@@ -1,84 +1,90 @@
-import { HypixelAPI } from "./HypixelAPI.js"
-import { SkyblockItems } from "./SkyblockItems.js"
-import fuzzy from "fuzzysort"
-import { LoggerCategory } from "../utils/Logger.js"
-import { deromanize } from "../utils/utils.js"
+import { HypixelAPI } from "./HypixelAPI";
+import { SkyblockItems } from "./SkyblockItems";
+import fuzzy from "fuzzysort";
+import { LoggerCategory } from "../utils/Logger";
+import { deromanize } from "../utils/utils";
 
 type BazaarProduct = {
-  id: string,
-  name: string,
-  sellSummary: Summary[],
-  buySummary: Summary[]
-  instasell?: number,
-  instabuy?: number
-}
+	id: string;
+	name: string;
+	sellSummary: Summary[];
+	buySummary: Summary[];
+	instasell?: number;
+	instabuy?: number;
+};
 
 type Summary = {
-  amount: number,
-  pricePerUnit: number,
-  orders: number
-}
+	amount: number;
+	pricePerUnit: number;
+	orders: number;
+};
 
 export class Bazaar {
+	private cachedProductList: BazaarProduct[] = [];
```

Comment:
If caching like this is a consistent pattern, maybe a `TTLCache` class should be implemented? i.e. a class that has a `get` and `set` method that respects TTL.